### PR TITLE
closes #380 feature trifid

### LINKED ISSUE #380: Trifid Cipher

### PR DESCRIPTION
**Pull request by t-m-z**

### CODE REVIEW

File: assets/i18n/en.json [hunk 1]
```diff
@@ -2985,6 +2985,17 @@
   "tapir_example" : "ABCDEFG → 05052 55565 78383",
   "tapir_online_help_url" : "https://blog.gcwizard.net/manual/en/tapir-substitution-cipher/02-how-does-gc-wizard-encrypt-or-decrypt-using-the-tapir-substitution/",
   "tapir_onetimepad" : "OTP Key",
+  "trifid_title" : "Trifid Chiffre",
```
Comment (S-Man42) [inline assets/i18n/en.json:2988]: In EN: Trifid Cipher

File: assets/i18n/fr.json [hunk 2]
```diff
@@ -2755,6 +2755,16 @@
   "tapir_description": "Code de l'ancienne armée de l'Allemagne de l'Est (NVA) et des services de renseignement (StaSi)",
   "tapir_example": "ABCDEFG → 05052 55565 78383",
   "tapir_onetimepad": "Clé OTP",
+  "trifid_title" : "Chiffrement Trifid",
```
Comment (S-Man42) [inline assets/i18n/fr.json:2758]: One thing you couldnt know about: With Ludo we decided to leave "Chiffrement" word to ensure that "Trifid" (or any other code) can be found under "T" and not everything under "C"

**Comment by S-Man42** [comment]
I checked with this website: https://cryptii.com/pipes/trifid-cipher

Same example as the their default, but Group Size 10. Is this a mistake on our side or on their side? I have no idea how this cipher works, so please check it :)

File: assets/i18n/en.json [hunk 3]
```diff
@@ -2985,6 +2985,17 @@
...
+  "trifid_description" : "Encryption based on three 3x3 Squares",
+  "trifid_example" : "HELLO → BQBRO",
+  "trifid_online_help_url" : "https://blog.gcwizard.net/manual/en/trifid-cipher/02-how-does-gc-wizard-encrypt-or-decrypt-using-the-trifid-cipher/",
+  "trifid_block_size" : "Blocksize",
```
Comment (S-Man42) [inline assets/i18n/en.json:2992]: "Block Size"

File: assets/i18n/en.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/en.json:2988]: done

File: assets/i18n/fr.json (hunk 2 above)
Comment (t-m-z) [reply assets/i18n/fr.json:2758]: done

File: assets/i18n/en.json (hunk 3 above)
Comment (t-m-z) [reply assets/i18n/en.json:2992]: done

**Comment by t-m-z** [comment]
Regarding your check with this website: https://cryptii.com/pipes/trifid-cipher and your finding "Same example as the their default, but Group Size 10. Is this a mistake on our side or on their side? I have no idea how this cipher works, so please check it :)"

The Wiki says:
The encryption protocol divides the plaintext into groups of fixed size (plus possibly one short group at the end): this confines encoding errors to the group in which they occur, an important consideration for ciphers that must be implemented by hand. The group size should be coprime to 3 to get the maximum amount of diffusion within each group: Delastelle gives examples with groups of 5 and 7 letters.

Hence the Block size or Group size can choosen freely

**Comment by t-m-z** [comment]
max-limit from integer spinner for block size is removed

**Comment by S-Man42** [comment]
Manually merged due to conflicts https://github.com/S-Man42/GCWizard/commit/e22f7ccbbb9bdc4e0e8f2d77cd34f82b4093af65

**Pull request closed without merging**